### DISCUSSION TITLE
Cannot make snippets work

### DISCUSSION THREAD

**Discussion by NFanoe** [squidfunk/mkdocs-material#7107] [Q&A]
I have `pymdownx.snippets` in the `markdown_extensions` section of my `mkdocs.yml`.

It just doesn't work. My other extensions work fine; snippets is the only one that fails. And yes, I've tried every combination [in the book](https://facelessuser.github.io/pymdown-extensions/extensions/snippets/).

Any advice? I know it's not much to go on ...

**Comment by kamilkrzyskow** [comment]
Hello @NFanoe,
a minimal example we could use as reproduction of "It just doesn't work" would be nice, together with a more thorough explanation how you run it and how it doesn't work (error, no effect), how does it fail exactly.
- https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/#creating-a-zip-file

This is a minimal example I made. [snippets-examaple.zip](https://github.com/squidfunk/mkdocs-material/files/15136781/snippets-examaple.zip)

Note that Snippets resolve paths from your current working directory aka the location where you run `mkdocs build`.
The default `base_path` for all snippets is `.` or a list containing only `.` to be exact. So in my example it looks for the `./snippets/snippet.md` file.
It would break if I would run `mkdocs build --config-file=some_directory/mkdocs.yml`, as the `./snippets/snippet.md` path would be incorrect, as `./some_directory/snippets/snippet.md` is what should be looked up, and Snippets doesn't handle such cases.

**Comment by NFanoe** [comment]
Thank you. The reason it's not working is probably that we have a complex setup (multi-repo and custom plugins). I'm guessing that's why the snippets folder can't reliably be found. (I know that the snippets extension is active because the file that includes the snippet hides the 
-- >8 -- thing when built.)
I'll get a developer on the case ... :)

**Reply by kamilkrzyskow** [reply]
If you have a developer, they could create a hook and decorate the [SnippetPreprocessor](https://github.com/facelessuser/pymdown-extensions/blob/8ed2fcb8ac688452ae1f5b29ab403e1e1d97e4e5/pymdownx/snippets.py#L45) class to modify the internal `self.base_path` to contain a proper absolute path after object init.
Similar to:
- https://github.com/squidfunk/mkdocs-material/discussions/7113

**Reply by NFanoe** [reply]
Ah. Good idea. Thanks!
